var/spack/repos/builtin/packages/intel-oneapi-compilers-classic/package.py

```diff
@@ -19,17 +19,19 @@ class IntelOneapiCompilersClassic(Package):
 
     has_code = False
 
-    phases = []
-
-    for ver in ['2022.1.0',
-                '2022.0.2',
-                '2022.0.1',
-                '2021.4.0',
-                '2021.3.0',
-                '2021.2.0',
-                '2021.1.2']:
+    # Versions before 2021 are in the `intel` package
+    # intel-oneapi versions before 2022 use intel@19.0.4
+    for ver, oneapi_ver in {
+        '2021.6.0': '2022.1.0:2022.2.0',
+    }.items():
         version(ver)
-        depends_on('intel-oneapi-compilers@' + ver, when='@' + ver, type='run')
+        depends_on('intel-oneapi-compilers@' + oneapi_ver, when='@' + ver, type='run')
+
+    @property
+    def oneapi_compiler_prefix(self):
+        oneapi_version = self.spec['intel-oneapi-compilers'].version
+        return self.spec['intel-oneapi-compilers'].prefix.compiler.join(
+            '%s' % oneapi_version).linux
```

Comment:
Why formatting versus `str()`?

Reply:
No good reason -- I changed how it worked several times and that's an artifact of an earlier version.